sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/PartitionKeyInternalHelper.java

```diff
@@ -92,6 +92,32 @@ static public String getEffectivePartitionKeyForHashPartitioningV2(PartitionKeyI
         }
     }
 
+    static String getEffectivePartitionKeyForMultiHashPartitioning(PartitionKeyInternal partitionKeyInternal) {
+        StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Can the size be pre-computed (with some bounds) to avoid re-sizing?

Reply:
Fixed as per suggestion